ext/afform/core/ang/af/afSearchParamSets.component.js

```diff
@@ -0,0 +1,258 @@
+      };
+
+      this.getSelectedParamSet = () => {
+        const hashParams = new URLSearchParams($window.location.hash.slice(1));
+        const urlValue = parseInt(hashParams.get('_s'));
+        return this.savedSets.find((searchParamSet) => searchParamSet.id === urlValue);
+      };
+
+      this.getCurrentParams = () => {
+        if (!this.afFieldset) {
+          return {};
+        }
+        const params = {};
+
+        params.filters = this.afFieldset.getFilterValues();
+
+        params.columns = {};
+
+        // find column toggle settings for any table search displays
+        // note 1: there can be multiple tables inside the same af-fieldset
+        // so we store columns keyed by display name
+        // note 2: this is unpleasant angular, but only reading data so
+        // shouldn't cause too many problems
+        $element.closest('[af-fieldset]').find('crm-search-display-table').each((i, $table) => {
+          const tableCtrl = angular.element($table).controller('crmSearchDisplayTable');
+          if (!tableCtrl) {
+            return;
+          }
+          const columns = tableCtrl.getToggledColumns();
+          if (!Object.keys(columns).length) {
+            return;
+          }
+          params.columns[tableCtrl.display] = columns;
+        });
+        return params;
+      };
+
+
+      this.renderSearchParamSetDetails = (paramSet) => {
+        const rendered = {};
+
+        if (paramSet.filters) {
+          // TODO: move meta getter to afFieldset?
+          const fieldMeta = this.afFieldset.getFieldMeta();
+
+          Object.keys(paramSet.filters).forEach((key) => {
+            const defn = fieldMeta[key];
+            const rawValue = paramSet.filters[key];
+
+            const formatValue = (v) => {
+              if (defn.options && defn.options.length) {
+                const selected = defn.options.find((o) => o.id == v);
+                if (selected) {
+                  return selected.label;
+                }
+              }
+              return v;
+            };
+
+            const value = (rawValue && rawValue.map) ? rawValue.map(formatValue).join(', ') : formatValue(rawValue);
+
+            rendered[defn.label] = value;
+          });
+        }
+
+        if (paramSet.columns) {
+          Object.keys(paramSet.columns).forEach((displayKey) => {
+            // TODO: how to get search label
+            const label = ts('%1 columns', {1: displayKey});
+            const columns = Object.values(paramSet.columns[displayKey]).join(', ');
+            rendered[label] = columns;
+          });
+        }
+
+        return rendered;
+      };
+
+      this.saveDialog = {
+        open: () => $element[0].querySelector('dialog.af-search-param-set-new').showModal(),
+        close: () => $element[0].querySelector('dialog.af-search-param-set-new').close(),
+        reset: () => {
+          this.saveDialog.label = ts('New search');
```

Comment:
IMO this would be better as a placeholder, but the field should otherwise be blank & required.